pkg/server/service/loadbalancer/wrr/wrr_test.go

```diff
@@ -112,3 +112,24 @@ func TestSticky(t *testing.T) {
 	assert.Equal(t, 0, recorder.save["first"])
 	assert.Equal(t, 3, recorder.save["second"])
 }
+
+func TestBalancerBias(t *testing.T) {
+	balancer := New(nil)
+
+	balancer.AddService("first", http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		rw.Header().Set("server", "first")
+		rw.WriteHeader(http.StatusOK)
+	}), Int(7))
+
+	balancer.AddService("second", http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		rw.Header().Set("server", "second")
+		rw.WriteHeader(http.StatusOK)
+	}), Int(3))
+
+	recorder := &responseRecorder{ResponseRecorder: httptest.NewRecorder(), save: map[string]int{}}
+	for i := 0; i < 4; i++ {
+		balancer.ServeHTTP(recorder, httptest.NewRequest(http.MethodGet, "/", nil))
+	}
+	assert.Less(t, recorder.save["first"], 4)
+	assert.Less(t, 0, recorder.save["second"])
+}
```

Comment:
I found that your example in the PR message was a good illustration of the problem, so I propose we change TestBalancerBias so that it represents perfectly that illustration. That way, people reading the git history can actually even better see what the problem was (and also we can avoid regressions). So, I propose:

```
// TestBalancerBias makes sure that the WRR algorithm spreads elements evenly
// right from the start, and that it does not "over-favor" the high-weighted ones
// with a biased start-up regime.
func TestBalancerBias(t *testing.T) {
	balancer := New(nil)

	balancer.AddService("first", http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
		rw.Header().Set("server", "A")
		rw.WriteHeader(http.StatusOK)
	}), Int(11))

	balancer.AddService("second", http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
		rw.Header().Set("server", "B")
		rw.WriteHeader(http.StatusOK)
	}), Int(3))

	recorder := &responseRecorder{ResponseRecorder: httptest.NewRecorder(), save: map[string]int{}}
	for i := 0; i < 14; i++ {
		balancer.ServeHTTP(recorder, httptest.NewRequest(http.MethodGet, "/", nil))
	}
	wantSequence := []string{"A", "A", "A", "B", "A", "A", "A", "A", "B", "A", "A", "A", "B", "A"}
	assert.Equal(t, wantSequence, recorder.sequence)
}
```

with the responseRecorder changed accordingly:

```
func (r *responseRecorder) WriteHeader(statusCode int) {
	r.save[r.Header().Get("server")]++
	r.sequence = append(r.sequence, r.Header().Get("server"))
	r.ResponseRecorder.WriteHeader(statusCode)
}
```

WDYT?

Reply:
Yeah, I agree with you.

Once I thought of adding such a test, but after the `ceil` trick, it changes every time, so the idea was abort. 
Since we have removed  the random trick from this LB algorithm, it would be perfect to add this. 

Fixing in minutes...